### ISSUE TITLE
Wrong inference for order of fields

### ISSUE THREAD

**Issue by curquiza** [meilisearch/MeiliSearch#494]
This problem appears in a random way.

Sometimes, when I create an index, and then, add documents, MeiliSearch does not keep the right order of fields.

### Version
On `master`, from [this commit](https://github.com/meilisearch/MeiliSearch/commit/72450c765d77222b18cef2bf5d149d1037603121).

**Comment by Kerollmops** [comment]
If this is only a visual bug: the searchable are returned in a random order to the user, therefore this line can be related.

https://github.com/meilisearch/MeiliSearch/blob/2b780ab2c5f2da2eb5e9475ff789da5cba11c359/meilisearch-http/src/routes/setting.rs#L293

Here we must collect into a `Vec` instead of an `HashSet` which final order is unknown.
@qdequele could you commit the change to the PR #496, please.

**Comment by Kerollmops** [comment]
We made tests with the fix I described above and it does not fix the original bug, when we add documents without an index the fields are not stored in the order they are sent. It is not related to the client request where the dictionary can expose keys in the wrong order as the JSON spec doesn't specify that keys are ordered.

We tried to directly set the searchable attributes before even sending a first document, this setting is an array, the JSON spec specify that the elements must always be ordered, but the HTTP server do not saved them in the same order as they were sent by the client. Notice that the order is constant, it as been saved in a wrong order and it is always the same wrong order in which it is displayed this is not a visual glitch.

I think that the bug is probably located in the `Schema` or the `FieldsMap` but I am not sure.
